### DISCUSSION TITLE
Fine-Tuning LLM with MacBook Pro M4 MAX

### DISCUSSION THREAD

**Discussion by softshipper** [ml-explore/mlx#1560] [General]
Hi everyone,

I have a question regarding the capability of the MacBook Pro M4 MAX with 128 GB RAM for fine-tuning large language model. Specifically, is this system sufficient to fine-tune LLaMA 3.2 with 3 billion parameters and using MLX?

Best regards

**Comment by awni** [comment]
It really depends on what you are trying to do. You can LoRA fine tune that model on a M1 with just 32Gb on a small to medium dataset. If you're dataset is larger, or has very long sequences, or you want to do full fine-tuning (train all the params, not low rank) you will probably want more RAM so 128GB will be better.

**Reply by softshipper** [reply]
Hi @awni,
Thank you for your response. I would like to fine-tune a model for a specific domain context - for example, a model specialized in selling tickets. Will this require a significant amount of computational power?

Best regards

**Reply by awni** [reply]
I think LoRA fine-tuning should work fine for that. And it should run fine especially on an M* Max laptop.

**Reply by softshipper** [reply]
Dear @awni,
One more question. After the fine-tuning can I still run the custom fine-tuned LLM on NVIDIA graphics card?

Best regards,
Anu

**Reply by awni** [reply]
- If you fine tune a 16-bit model (and fuse the adapters into it) it should work in most cases
- For 4-bit it has to be dequantized to work

**Reply by softshipper** [reply]
> it should work in most cases

But what does _in most cases_ mean? Does this imply that it might not work sometimes?

**Reply by awni** [reply]
Yes it might not work sometimes. MOEs are an example of when it doesn’t work.

**Reply by awni** [reply]
In general you need to convert the model back to the Transformers format. When the conversion is identity it works when it’s not it doesn’t. This is on a model by model basis.

**Reply by softshipper** [reply]
@awni Thank you for the clarification.
